src/tools/runk/README.md

```diff
@@ -0,0 +1,277 @@
+|-----------------------|-------------------------|-------|-------|
+| time [msec]           | 39.01                   | 53.09 | 35.35 |
+| memory footprint [MB] | 4.223                   | 16.54 | 3.157 |
+
+
+## Status of runk
+
+`runk` is currently an experimental tool.
+We drafted the initial code here, and any contributions to `runk` and [`kata-agent`](https://github.com/kata-containers/kata-containers/tree/main/src/agent) are welcome.
+
+Regarding features compared to `runc`, see the `Status of runk` section in the [issue](https://github.com/kata-containers/kata-containers/issues/2784).
+
+## Building
+
+> **Warnings:**
+> The runk is experimental.
+> Only continue if you are using a non-critical system.
+
+`runk` uses the modified the `kata-agent` binary, `oci-kata-agent`, which is an agent to be called from `runk`.
+Therefore, you also need to build the `oci-kata-agent` to run `runk`.
+
+You can build both `runk` and `oci-kata-agent` as follows.
+
+```bash
+$ cd runk
+$ make
+```
+
+To install `runk` and `oci-kata-agent` into default directory for install executable program (`/usr/local/bin`):
+
+```bash
+$ sudo make install
+```
+
+## Using `runk` directly
+
+Please note that `runk` is a low level tool not developed with an end user in mind.
+It is mostly employed by other higher-level container software like `containerd`.
+
+If you still want to use `runk` directly, here's how.
+
+### Prerequisites
+
+It is necessary to create an OCI bundle to use the tool. The simplest method is:
+
+``` bash
+$ bundle_dir="bundle"
+$ rootfs_dir="$bundle_dir/rootfs"
+$ image="busybox"
+$ mkdir -p "$rootfs_dir" && (cd "$bundle_dir" && runk spec)
+$ sudo docker export $(sudo docker create "$image") | tar -C "$rootfs_dir" -xvf -
+```
+
+> **Note:**
+> If you use the unmodified `runk spec` template, this should give a `sh` session inside the container.
+> However, if you use `runk` directly and run a container with the unmodified template,
+> `runk` cannot launch the `sh` session because `runk` does not support terminal handling yet.
+> You need to edit the process field in the `config.json` should look like this below
+> with `"terminal": false` and `"args": ["sleep", "10"]`.
+
+```json
+"process": {
+    "terminal": false,
+    "user": {
+        "uid": 0,
+        "gid": 0
+    },
+    "args": [
+        "sleep",
+        "10"
+    ],
+    "env": [
+        "PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+        "TERM=xterm"
+    ],
+    "cwd": "/",
+    [...]
+}
+```
+
+If you want to launch the `sh` session inside the container, you need to run `runk` from `containerd`.
+
+Please refer to the [Using runk from containerd](#Using-runk-from-containerd) section
+
+### Running a container
+
+Now you can go through the [lifecycle operations](https://github.com/opencontainers/runtime-spec/blob/master/runtime.md)
+in your shell.
+
+```bash
+$ cd $bundle_dir
+
+# Create a container
+$ sudo runk create test
```

Comment:
I think you should explain why it is necessary to run `runk` as `root`.

Reply:
I fixed it.